### ISSUE TITLE
VirtualizingStackPanel cannot work with ScaleTransform animation.

### ISSUE THREAD

**Issue by kyuranger** [AvaloniaUI/Avalonia#13437]
### **Describe the bug**
VirtualizingStackPanel cannot work with ScaleTransform animation.

### **To Reproduce**
1.XAML:
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication1.MainWindow"
		xmlns:Local="using:AvaloniaApplication1"
        Title="AvaloniaApplication1">
	<Window.Styles>
		<Style Selector="Grid.Expand">
			<Style.Animations>
				<Animation Duration="0:0:0.3" FillMode="Forward">
					<Animation.Easing>
						<CubicEaseOut></CubicEaseOut>
					</Animation.Easing>
					<KeyFrame Cue="0%">
						<Setter Property="ScaleTransform.ScaleY" Value="0"/>
					</KeyFrame>
					<KeyFrame Cue="100%">
						<Setter Property="ScaleTransform.ScaleY" Value="1"/>
					</KeyFrame>
				</Animation>
			</Style.Animations>
		</Style>
	</Window.Styles>
	<Grid RowDefinitions="auto,auto">
		<Button Name="B">Show</Button>
		<Grid RowDefinitions="auto,auto" Name="G" Grid.Column="1" IsVisible="false">
			<Border Background="Red" Height="200" Width="350"></Border>
			<ItemsControl ClipToBounds="True" Grid.Row="1">
				<ItemsControl.ItemsPanel>
					<ItemsPanelTemplate>
						<VirtualizingStackPanel></VirtualizingStackPanel>
					</ItemsPanelTemplate>
				</ItemsControl.ItemsPanel>
				<ItemsControl.ItemTemplate>
					<DataTemplate>

					</DataTemplate>
				</ItemsControl.ItemTemplate>
			</ItemsControl>
		</Grid>
	</Grid>
</Window>
```

2.Code-behind:
```
using Avalonia.Controls;
using Avalonia.Media;

namespace AvaloniaApplication1
{
    public partial class MainWindow : Window
    {
        public MainWindow()
        {
            InitializeComponent();
            B.Click += B_Click;
        }

        private void B_Click(object? sender, Avalonia.Interactivity.RoutedEventArgs e)
        {
            G.IsVisible = true;
            if (G.RenderTransform == null)
            {
                G.RenderTransform = new ScaleTransform(1, 0);
            }
            G.Classes.Clear();
            G.Classes.Add("Expand");
        }
    }
}
```

3This is what appears when the button is clicked:
![1](https://github.com/AvaloniaUI/Avalonia/assets/30720036/59286ca6-0c13-41c5-bbef-7905a7032860)

As you see, it doesn't play animation any. And in DevTools it will throws error about RenderTransform:
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/ccff7c15-b6a9-4125-bf78-0fd93afd5743)

PS:To reproduce this problem, you do not even need to assign a value to the ItemSource of ItemsControl.

### **Expected behavior**

However, if I replace the VirtualizingStackPanel with StackPanel, it works well and it is what I expect.

![2](https://github.com/AvaloniaUI/Avalonia/assets/30720036/d837138a-4595-4e1c-806e-e256630518a4)

**Environment**
OS: Windows 10 19045.3570
Avalonia-Version: 11.0.5

**Comment by kyuranger** [comment]
I found a temporary solution. When playing the ScaleTransform animation,set IsVisible of ItemsControl to false, and then set it to true after the animation plays.

**Comment by timunie** [comment]
@kyuranger I guess this is by design. Because virtulization Stackpanel needs "valid" bounds that are neither 0 nor infinite. otherwise it cannot caculate it's childrens correctly.

**Comment by kyuranger** [comment]
> @kyuranger I guess this is by design. Because virtulization Stackpanel needs "valid" bounds that are neither 0 nor infinite. otherwise it cannot caculate it's childrens correctly.

I think what you said makes sense, but WPF doesn't seem to have such a problem (I discovered this problem just by migrating my old WPF project to Avalonia). And I think that even if Avalonia cannot solve the problem of VirtualizingStackPanel, it should skip rending VirtualizingStackPanel in animation instead of not rendering all content.

**Comment by MrJul** [comment]
This sample in the OP now works fine with the latest 11.1 nightly build (and fails in 11.0.5). The fix is likely #13188.

I'm closing this issue as fixed, please reopen if needed.
